rfd/0001-tooling-requirements.md

```diff
@@ -0,0 +1,150 @@
+
+#### CD workflow
+This workflow should run when a PR is merged, and optionally upon PR commits as a way to produce test builds. When ran, this workflow should upload artifacts as Github Packages where possible. These artifacts should not be uploaded to AWS unless absolutely necessary, and with Security's permission, as outlined in the [Security considerations section](#security-considerations). The name of these artifacts should include a dev tag generated by `git describe --tags` to ensure uniqueness.
+
+#### Release workflow
+The release workflow should be triggered when a new version is added to the CHANGELOG on `main` branch. When this occurs, the workflow should:
+1. Extract the new version from the CHANGELOG.
+2. Create a new Git tag targeting the commit that modified the CHANGELOG, with name `v<new version>-<project name>` with all separator characters (e.g. spaces and underscores) separated by dashes. This tag should be parsable by [this npm package](https://www.npmjs.com/package/semver) that Renovate uses for semver coercion.
+3. Build a new release of the tool, following the general logic of the CD workflow. Either the release version or Git tag may be used where appropriate.
+4. Create a new release, named after the Git tag, that includes the changes from the CHANGELOG. The release should also either include the built artifacts, or instructions on retrieving them (such as `docker pull ghcr.io/gravitational/shared-workflows/some-project:v1.2.3`).
+
+### Project releases
+The process for releasing a new version of a project should be as follows:
+1. PRs are filed to add/modify the project.
+2. The project's CI workflow triggers to validate the changes, codeowners approve, and PRs are merged.
+3. Artifacts are built for each PR to facilitate release testing.
+4. A new PR is filed and merged that updates the changelog to a new version.
+5. The release workflow is triggered, which creates a new tag and publishes a fresh build.
+
+### Dependency management
+Historically this repo has used Dependabot to manage dependency updates. Unfortunately however, Dependabot does not support two key "languages" used by Github Actions: actions themselves (used when building composite actions), and Dockerfiles (used when building Dockerfile actions). As a result, [Renovate](https://docs.renovatebot.com/) will be used for new and migrated projects instead. Renovate is highly flexible and can be configured on a per-project basis.
+
+Renovate will self-hosted and configured to run every 12 hours. Updates for each project should be grouped as much as reasonably possible to reduce the number of PRs opened at any given time. Update PRs will be assigned to code owners for each project. Digest pinning will be used wherever supported.
```

Comment:
```suggestion
Renovate will self-hosted and configured to run daily. Updates for each project should be grouped as much as reasonably possible to reduce the number of PRs opened at any given time. Update PRs will be assigned to code owners for each project. Digest pinning will be used wherever supported.
```

I don't think we'd realistically merge more than once a day anyhow.  In `teleport`/`teleport.e` we do updates weekly to cut down on sustaining -- and I'm starting to lean that way for `cloud-terraform`.  In my opinion: the time lag for picking up a critical security update is lower risk to Gravitational than the cost of having maintainers spend more time on sustaining dependency updates and less time on new development.